rllib/models/tf/recurrent_net.py

```diff
@@ -203,22 +210,35 @@ def forward(self, input_dict: Dict[str, TensorType],
 
         # Concat. prev-action/reward if required.
         prev_a_r = []
+        # Prev actions: Try flattening into 1D array (one-hot discrete
+        # components). This only properly works with complex action spaces
+        # iff action flattening has been disabled.
         if self.model_config["lstm_use_prev_action"]:
             prev_a = input_dict[SampleBatch.PREV_ACTIONS]
-            if isinstance(self.action_space, (Discrete, MultiDiscrete)):
-                prev_a = one_hot(prev_a, self.action_space)
-            prev_a_r.append(
-                tf.reshape(tf.cast(prev_a, tf.float32), [-1, self.action_dim]))
+            if self.model_config["_disable_action_flattening"]:
+                prev_a_r.append(
+                    flatten_inputs_to_1d_tensor(
+                        prev_a,
+                        spaces_struct=self.action_space_struct,
+                        time_axis=False,
+                    ))
+            else:
+                if isinstance(self.action_space, (Discrete, MultiDiscrete)):
```

Comment:
can you also commnet here :)
we only support Discrete and MultiDiscrete here, everything else is "complex action spaces" that you mentioned above right?